src/types/character_attributes_iter.rs

```diff
@@ -0,0 +1,87 @@
+use crate::models::character::Character;
+
+#[derive(Debug, Clone)]
+pub struct CharacterAttributesIter<'a>{
+    character: &'a Character,
+    index: usize,
+}
+
+impl <'a> CharacterAttributesIter<'a> {
+    pub fn new(character: &'a Character)->Self {
+        Self{
+            character,
+            index: 0
+        }
+    }
+}
+
+pub struct CharacterAttribute {
+    pub name: &'static str,
+    pub level: u32
+}
```

Comment:
Ain't it a model? If so, move it to separate file in `models` directory. Otherwise remove access specifier:
```suggestion
struct CharacterAttribute {
    pub name: &'static str,
    pub level: u32
}
```

Reply:
As of now it's the only way to avoid writing another macro, it's a warpper type which allows to address fields by identifier in `create_table!` macro. A change to model attributes from raw u32 fields to a slice addressable by BaseAttribute enum would allow for it's removal